dashboard/app/helpers/aichat_openai_helper.rb

```diff
@@ -71,6 +71,29 @@ def self.request_chat_completion(messages, temperature)
     [response, body&.dig('usage')]
   end
 
+  def self.request_structured_chat_completion(messages, temperature, response_schema)
+    http_response = client.request_chat_completion(
+      messages,
+      temperature,
+      extra: {
+        response_format: {
+          type: "json_schema",
+          json_schema: response_schema
+        }
+      }
+    )
+    body = JSON.parse(http_response.body)
+    raise StandardError.new(body['error']) if body['error']
+    # Check if the returned content is a json, else throw error
+    response_json = body.dig("choices", 0, "message", "content")
```

Comment:
Is this error checking different than what we're doing in `request_chat_completion` above?